cloud/services/virtualmachines/virtualmachines.go

```diff
@@ -109,6 +114,7 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
 
 	virtualMachine := compute.VirtualMachine{
 		Location: to.StringPtr(s.Scope.Location()),
+		Plan:     plan,
```

Comment:
so by default go sdk does a nil instead of an empty struct. Can we please follow the same, and return nil instead of empty with generateImagePlan?

Reply:
sounds good.

This is temporary btw, I'm hoping to be able to remove this code once the image becomes 1st party.